In [8]:
import pandas as pd

data = pd.read_csv('https://raw.githubusercontent.com/fahmimnalfrzki/Dataset/main/Mall%20Customers.csv')
data.head()

Unnamed: 0  CustomerID  Gender  Age  Annual Income (k$)  \
0           0           1    Male   19                  15   
1           1           2    Male   21                  15   
2           2           3  Female   20                  16   
3           3           4  Female   23                  16   
4           4           5  Female   31                  17   

   Spending Score (1-100)  Label  
0                      39      0  
1                      81      4  
2                       6      0  
3                      77      4  
4                      40      0

Statistik dasar dari data customer mall:

In [10]:
data.describe().T

count    mean        std   min    25%    50%     75%  \
Unnamed: 0              200.0   99.50  57.879185   0.0  49.75   99.5  149.25   
CustomerID              200.0  100.50  57.879185   1.0  50.75  100.5  150.25   
Age                     200.0   38.85  13.969007  18.0  28.75   36.0   49.00   
Annual Income (k$)      200.0   60.56  26.264721  15.0  41.50   61.5   78.00   
Spending Score (1-100)  200.0   50.20  25.823522   1.0  34.75   50.0   73.00   
Label                   200.0    2.19   1.208804   0.0   1.00    3.0    3.00   

                          max  
Unnamed: 0              199.0  
CustomerID              200.0  
Age                      70.0  
Annual Income (k$)      137.0  
Spending Score (1-100)   99.0  
Label                     4.0

In [20]:
df[
    (df['year_id'] > 1950) &
    (df['fran_id'] == 'Knicks') &
    (df['pts'] > 150)
]

SyntaxError: invalid syntax (Temp/ipykernel_28284/3643989644.py, line 7)

- Rata-rata usia pelanggan mall adalah sekitar 38.85 tahun dengan usia paling muda adalah 18 tahun dan yang paling tua 70 tahun.
- Rata-rata pendapatan tahunan pelanggan adalah 60,560 dengan pendapatan paling rendah 15000 dan paling tinggi 137000.
- Spending score tiap pelanggan bervariasi dari 1 hingga 99 (tidak ada yang dapat nilai sempurna) dengan nilai rata-rata 50,2.

Label 0 memiliki rata-rata spending score kecil, annual income relatif kecil dibanding label lainnya, dan usia rata-rata 45 tahun. Kesempatan untuk target pemasaran di label ini relatif kecil.

In [11]:
label_0 = data[data["Label"] == 0]
label_0.describe().T

count       mean        std   min   25%   50%   75%  \
Unnamed: 0               23.0  22.000000  13.564660   0.0  11.0  22.0  33.0   
CustomerID               23.0  23.000000  13.564660   1.0  12.0  23.0  34.0   
Age                      23.0  45.217391  13.228607  19.0  35.5  46.0  53.5   
Annual Income (k$)       23.0  26.304348   7.893811  15.0  19.5  25.0  33.0   
Spending Score (1-100)   23.0  20.913043  13.017167   3.0   9.5  17.0  33.5   
Label                    23.0   0.000000   0.000000   0.0   0.0   0.0   0.0   

                         max  
Unnamed: 0              44.0  
CustomerID              45.0  
Age                     67.0  
Annual Income (k$)      39.0  
Spending Score (1-100)  40.0  
Label                    0.0

Label 1 merepresentasikan pelanggan dengan annual income tinggi, spending score tinggi, dan memiliki rentang umur menengah. Label 1 rata-rata spending score paling tinggi dibanding label lainnya. Annual income relatif tinggi dibanding lainnya. Kesempatan yang bagus untuk target pemasaran. Dengan karakter ini dan jumlah anggota klusternya yang relatif tinggi, label ini merupakan target market yang paling baik. Produk yang cocok untuk label ini adalah produk-produk orang dewasa usia 32 tahunan. Melakukan pengadaan produk baru yang sering dibeli oleh masyarakat dengan karakter label ini adalah ide yang dapat dipertimbangkan. Perubahan harga produk (yang masih rasional dan sesuai dengan kualitas produk tentunya) tidak akan terlalu mempengaruhi daya beli dan daya tarik label ini.

In [13]:
label_1 = data[data["Label"] == 1]
label_1.describe().T

count        mean        std    min    25%    50%  \
Unnamed: 0               39.0  161.000000  22.803509  123.0  142.0  161.0   
CustomerID               39.0  162.000000  22.803509  124.0  143.0  162.0   
Age                      39.0   32.692308   3.728650   27.0   30.0   32.0   
Annual Income (k$)       39.0   86.538462  16.312485   69.0   75.5   79.0   
Spending Score (1-100)   39.0   82.128205   9.364489   63.0   74.5   83.0   
Label                    39.0    1.000000   0.000000    1.0    1.0    1.0   

                          75%    max  
Unnamed: 0              180.0  199.0  
CustomerID              181.0  200.0  
Age                      35.5   40.0  
Annual Income (k$)       95.0  137.0  
Spending Score (1-100)   90.0   97.0  
Label                     1.0    1.0

Label 2 memiliki rata-rata pendapatan tinggi dibanding label lainnya, tetapi nilai spending score-nya paling rendah. Usia rata-rata label ini sedikit di atas rata-rata (40 tahun). Orang yang sedang berhemat. Trick promo produk beli 1 gratis 1 mungkin dapat merangsang daya tarik anggota label untuk memberikan lebih banyak penghasilan mereka (yang relatif paling tinggi daripada label lainnya) untuk berbelanja.

In [14]:
label_2 = data[data["Label"] == 2]
label_2.describe().T

count        mean        std    min    25%    50%  \
Unnamed: 0               37.0  161.945946  21.743385  124.0  144.0  162.0   
CustomerID               37.0  162.945946  21.743385  125.0  145.0  163.0   
Age                      37.0   40.324324  11.525465   19.0   34.0   41.0   
Annual Income (k$)       37.0   87.432432  16.272916   70.0   77.0   81.0   
Spending Score (1-100)   37.0   18.189189  10.715611    1.0   10.0   16.0   
Label                    37.0    2.000000   0.000000    2.0    2.0    2.0   

                          75%    max  
Unnamed: 0              180.0  198.0  
CustomerID              181.0  199.0  
Age                      47.0   59.0  
Annual Income (k$)       97.0  137.0  
Spending Score (1-100)   26.0   40.0  
Label                     2.0    2.0

Label 3 merupakan label dengan jumlah anggota paling banyak dengan spending score dan annual income sedang (mendekati rata-rata) dan usia rata-rata 43 tahun. Target pemasaran untuk orang-orang tua.

In [15]:
label_3 = data[data["Label"] == 3]
label_3.describe().T

count       mean        std   min   25%   50%    75%  \
Unnamed: 0               79.0  84.012658  23.061508  43.0  64.5  84.0  103.5   
CustomerID               79.0  85.012658  23.061508  44.0  65.5  85.0  104.5   
Age                      79.0  43.126582  16.446781  18.0  27.0  47.0   54.5   
Annual Income (k$)       79.0  54.822785   8.576592  39.0  48.0  54.0   62.0   
Spending Score (1-100)   79.0  49.835443   6.276595  35.0  45.5  50.0   55.0   
Label                    79.0   3.000000   0.000000   3.0   3.0   3.0    3.0   

                          max  
Unnamed: 0              126.0  
CustomerID              127.0  
Age                      70.0  
Annual Income (k$)       71.0  
Spending Score (1-100)   61.0  
Label                     3.0

Label 4 merupakan label dengan spending score dan pendapatan sedang tetapi usia rata-rata membernya adalah anak muda usia 26 tahunan, tapi banyak terdiri dari masyarakat usia lebih muda. Label ini bagus untuk target pasar anak-anak muda.


In [16]:
label_4 = data[data["Label"] == 4]
label_4.describe().T

count       mean        std   min    25%   50%    75%  \
Unnamed: 0               22.0  22.090909  13.147185   1.0  11.50  22.0  32.50   
CustomerID               22.0  23.090909  13.147185   2.0  12.50  23.0  33.50   
Age                      22.0  25.272727   5.257030  18.0  21.25  23.5  29.75   
Annual Income (k$)       22.0  25.727273   7.566731  15.0  19.25  24.5  32.25   
Spending Score (1-100)   22.0  79.363636  10.504174  61.0  73.00  77.0  85.75   
Label                    22.0   4.000000   0.000000   4.0   4.00   4.0   4.00   

                         max  
Unnamed: 0              45.0  
CustomerID              46.0  
Age                     35.0  
Annual Income (k$)      39.0  
Spending Score (1-100)  99.0  
Label                    4.0

Label 3 dan 4 merupakan label yang harus dapat dipertahankan oleh pihak Mall. Tidak ada salahnya untuk mencoba memperkenalkan produk baru ke label ini (tentu saja produk yang sesuai dengan karakter usia mereka). Namun perlu diperhatikan bahwa mungkin penetapan dan sedikit perubahan harga produk yang label ini beli dapat mempengaruhi daya beli mereka.